counsel.el

```diff
@@ -1678,6 +1679,57 @@ When INITIAL-INPUT is non-nil, use it in the minibuffer during completion."
           (format "http://debbugs.gnu.org/cgi/bugreport.cgi?bug=%s"
                   (substring url 1)))))))
 
+(defvar counsel-url-expansions
+  nil
+  "List of (REGEXP . FORMAT) pairs.
+
+`counsel-url-expand' will expand the word at point according to
+FORMAT for the first matching REGEXP.  FORMAT can be either a
+string or a function.  If it is a string, it will be used as the
+format string for the `format' function, with the word at point
+as the next argument.  If it is a function, it will be called
+with the word at point as the sole argument.
+
+For example, a pair of the form:
+  '(\"^BSERV-[0-9]+$\" . \"https://jira.atlassian.com/browse/%s\")
+will expand to URL `https://jira.atlassian.com/browse/BSERV-100'
+when the word at point is BSERV-100.
+
+If the format element is a function, more powerful
+transformations are possible.  As an example,
+  '(\"^issue\\([0-9]+\\)$\" .
+    (lambda (word)
+      (concat \"http://debbugs.gnu.org/cgi/bugreport.cgi?bug=\"
+              (match-string 1 word))))
+trims the \"issue\" prefix from the word at point before creating the URL.")
+
+(defun compat-seq-some (pred lst)
+  "Call PRED on each element of LST and return first non-nil result.
+\(Implementation of seq-some for Emacs prior to version 25.)"
+  (catch 'result
+    (dolist (elt lst)
+      (let ((result (funcall pred elt)))
+        (when result
+          (throw 'result result))))
+    nil))
```

Comment:
You can just use `cl-some` instead of `seq-some`. FYI, global definitions should be prefixed with the package name by [Elisp convention](https://www.gnu.org/software/emacs/manual/html_node/elisp/Coding-Conventions.html).

Reply:
ok, thanks.